https://bugs.gentoo.org/930060
https://gitlab.com/gnuwget/wget/-/issues/19
https://gitlab.com/gnuwget/wget/-/merge_requests/39

From 5f0aa59239c36fc945b94d8ab91562d56e5bf776 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Tue, 16 Apr 2024 03:18:40 +0100
Subject: [PATCH] Fix libproxy build with --disable-debug

The definition of debug_logprintf in src/log.c is guarded by ENABLE_DEBUG
(although its prototype is unconditionally available in src/log.h).

The uses of debug_logprintf in src/retr.c aren't guarded by ENABLE_DEBUG.

Use the DEBUGP macro which is designed for this purpose.

* src/retr.c (getproxy): Use DEBUGP macro.

Fixes: https://gitlab.com/gnuwget/wget/-/issues/19
Copyright-paperwork-exempt: Yes
--- a/src/retr.c
+++ b/src/retr.c
@@ -1498,21 +1498,21 @@ getproxy (struct url *u)
       pxProxyFactory *pf = px_proxy_factory_new ();
       if (!pf)
         {
-          debug_logprintf ("Allocating memory for libproxy failed");
+          DEBUGP (("Allocating memory for libproxy failed"));
          return NULL;
         }
 
-      debug_logprintf ("asking libproxy about url '%s'\n", u->url);
+      DEBUGP (("asking libproxy about url '%s'\n", u->url));
       char **proxies = px_proxy_factory_get_proxies (pf, u->url);
       if (proxies)
         {
           if (proxies[0])
             {
-              debug_logprintf ("libproxy suggest to use '%s'\n", proxies[0]);
+              DEBUGP (("libproxy suggest to use '%s'\n", proxies[0]));
               if (strcmp (proxies[0], "direct://") != 0)
                 {
                   proxy = xstrdup (proxies[0]);
-                  debug_logprintf ("libproxy setting to use '%s'\n", proxy);
+                  DEBUGP (("libproxy setting to use '%s'\n", proxy));
                 }
             }
 
-- 
GitLab
